src/Symfony/Component/Messenger/Command/FailedMessagesShowCommand.php

```diff
@@ -121,6 +131,38 @@ private function listMessages(SymfonyStyle $io, int $max)
         $io->comment('Run <comment>messenger:failed:show {id} -vv</comment> to see message details.');
     }
 
+    private function listMessagesPerClass(SymfonyStyle $io)
+    {
+        /** @var ListableReceiverInterface $receiver */
+        $receiver = $this->getReceiver();
+        $envelopes = $receiver->all();
+
+        $countPerClass = [];
+
+        foreach ($envelopes as $envelope) {
+            if (!isset($countPerClass[\get_class($envelope->getMessage())])) {
+                $countPerClass[\get_class($envelope->getMessage())] = 0;
```

Comment:
you could use `$countPerClass[$c = \get_class($envelope->getMessage())] = [$c, 0];` 
to avoid the `array_walk` below